flixel/system/FlxBasePreloader.hx

```diff
@@ -199,45 +221,137 @@ class FlxBasePreloader extends NMEPreloader
 	 */
 	private function checkSiteLock():Void
 	{
-		#if web
-		if (!_urlChecked && (allowedURLs != null))
+        #if web
+        if (_urlChecked) return;
+
+        if (!isHostUrlAllowed())
 		{
-			if (!isHostUrlAllowed())
-			{
-				var tmp = new Bitmap(new BitmapData(stage.stageWidth, stage.stageHeight, true, FlxColor.WHITE));
-				addChild(tmp);
-				
-				var format = new TextFormat();
-				format.color = 0x000000;
-				format.size = 16;
-				format.align = TextFormatAlign.CENTER;
-				format.bold = true;
-				format.font = "system";
-				
-				var textField = new TextField();
-				textField.width = tmp.width - 16;
-				textField.height = tmp.height - 16;
-				textField.y = 8;
-				textField.multiline = true;
-				textField.wordWrap = true;
-				textField.defaultTextFormat = format;
-				textField.text = "Hi there!  It looks like somebody copied this game without my permission.  Just click anywhere, or copy-paste this URL into your browser.\n\n" + allowedURLs[0] + "\n\nto play the game at my site.  Thanks, and have fun!";
-				addChild(textField);
-				
-				textField.addEventListener(MouseEvent.CLICK, goToMyURL);
-				tmp.addEventListener(MouseEvent.CLICK, goToMyURL);
-				
-				removeEventListener(Event.ENTER_FRAME, onEnterFrame);
-			}
-			else
-			{
-				_urlChecked = true;
-			}
-		}
-		#end
+            removeChildren();
+            removeEventListener(Event.ENTER_FRAME, onEnterFrame);
+
+			createSiteLockFailureScreen();
+        }
+		else
+		{
+            _urlChecked = true;
+        }
+        #end
 	}
 	
 	#if web
+	/**
+	 * When overridden, allows the customized creation of the sitelock failure screen.
+	 */
+	private function createSiteLockFailureScreen():Void
+	{
+		addChild(createSiteLockFailureBackground(0xffffff, 0xe5e5e5));
+		addChild(createSiteLockFailureIcon(0xe5e5e5, 0.9));
+		addChild(createSiteLockFailureText(30));
+	}
+
+    private function createSiteLockFailureBackground(innerColor:UInt, outerColor:UInt):Shape
+	{
```

Comment:
Sorry about this.  VSCode normally does the right thing by matching the formatting of the file you are editing.  However, I use spaces in my own code, so any new files that I write will use spaces instead of tabs.  Since I had to do a big copy-paste job for this one, the spaces I normally used leaked in here.  I thought I caught that and normalized it, but I think that may have only been for a few lines here and there.